libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -550,8 +550,11 @@ void lua_scripts::run(void) {
             if ((_debug_options.get() & uint8_t(DebugLevel::RUNTIME_MSG)) != 0) {
                 GCS_SEND_TEXT(MAV_SEVERITY_DEBUG, "Lua: Running %s", scripts->name);
             }
-            // copy name for logging, cant do it after as script reschedule moves the pointers
-            const char * script_name = scripts->name;
+            // take a copy of the script name for the purposes of
+            // logging statistics.  "scripts" may become invalid
+            // during the "run_next_script" call, below.
+            char script_name[128+1] {};
```

Comment:
We only log to a 16 char space, I don't think you need to copy the full thing.

Reply:
> We only log to a 16 char space, I don't think you need to copy the full thing.

Yeah, I thought that too.  But the  trimming of the directory off the path happens closer to where we log it.